consensus/istanbul/backend/announce.go

```diff
@@ -246,6 +256,20 @@ func (sb *Backend) pruneAnnounceDataStructures() error {
 		return err
 	}
 
+	sb.lastSignedAnnounceVersionsGossipedMu.Lock()
+	for remoteAddress := range sb.lastSignedAnnounceVersionsGossiped {
+		if !validatorConnSet[remoteAddress] {
+			logger.Trace("Deleting entry from lastSignedAnnounceVersionsGossiped", "address", remoteAddress, "gossip timestamp", sb.lastSignedAnnounceVersionsGossiped[remoteAddress])
+			delete(sb.lastSignedAnnounceVersionsGossiped, remoteAddress)
```

Comment:
Because this is a DDoS protection with a timeout, we should keep the entry for the address in this table until that timeout has expired.

Reply:
I recently changed the pruning to happen every 10 minutes, so I think this is a little less of a worry. Given that this isn't very critical would you be okay with treating this as a future optimization in a later PR?

Reply:
This change is as simple as `if time.Since(sb.lastAnnounceGossiped[remoteAddress]) < signedAnnounceVersionGossipCooldownDuration { continue }` so we should esither include this line now or not plan on it. I agree that it's not so important, so if you disagree with my reasoning, feel free to ignore it.

Reply:
nvm I'll just add it